### PR TITLE
Support for HTI21D family sensor on I2C bus

### PR DESCRIPTION
**Pull request by dianlight**
Support for HTI21D family sensor on I2C bus 

- Also include fix i2c read with 0 length write on linux_mcu as discussed in #2795 

Signed-off-by: Lucio Tarantino <lucio.tarantino@gmail.com>

### CODE REVIEW

File: klippy/extras/htu21d.py [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+#       Si7013 - Untested
+#       Si7020 - Untested
+#       Si7021 - Untested
+#       SHT21  - Untested
+#
+# NOTE: Implementation need the support od 0 length send before read on MCU
+#
+# Copyright (C) 2020  Lucio Tarantino <lucio.tarantino@gmail.com>
+#
+# This file may be distributed under the terms of the GNU GPLv3 license.
```
Comment (KevinOConnor) [inline klippy/extras/htu21d.py:15]: Please change this so the location of the copyright is similar to other files:
```
# One line description
#
# Copyright ...
# 
# This file may be distributed under the terms of the GNU GPLv3 license.

# Any other comments as needed...
```

File: src/linux/i2c.c [hunk 2]
```diff
@@ -91,7 +91,8 @@ void
 i2c_read(struct i2c_config config, uint8_t reg_len, uint8_t *reg
          , uint8_t read_len, uint8_t *data)
 {
-    i2c_write(config, reg_len, reg);
+    if( reg_len != 0)
```
Comment (KevinOConnor) [inline src/linux/i2c.c:94]: White space error.

Comment (dianlight) [reply src/linux/i2c.c]: Done in commit: f5f6bb3

File: klippy/extras/htu21d.py (hunk 1 above)
Comment (dianlight) [reply klippy/extras/htu21d.py]: Done in commit 2cd7404

**Comment by KevinOConnor** [comment]
Thanks.

-Kevin

**Pull request merged**